cmd/notary/delegations.go

```diff
@@ -0,0 +1,172 @@
+	if err != nil {
+		fatalf(err.Error())
+	}
+
+	delegationRoles, err := nRepo.GetDelegationRoles()
+	if err != nil {
+		return fmt.Errorf("Error retrieving delegation roles for repository %s: %v", gun, err)
+	}
+
+	cmd.Println("")
+	prettyPrintRoles(delegationRoles, cmd.Out())
+	cmd.Println("")
+	return nil
+}
+
+// delegationRemove removes a public key from a specific role in a GUN
+func (d *delegationCommander) delegationRemove(cmd *cobra.Command, args []string) error {
+	if len(args) != 3 {
+		return fmt.Errorf("must specify the Global Unique Name, the Key ID and the role of the delegation to remove")
+	}
+
+	config := d.configGetter()
+
+	gun := args[0]
+	keyID := args[1]
+	role := args[2]
+
+	// no online operations are performed by add so the transport argument
+	// should be nil
+	nRepo, err := notaryclient.NewNotaryRepository(config.GetString("trust_dir"), gun, getRemoteTrustServer(config), nil, retriever)
+	if err != nil {
+		fatalf(err.Error())
+	}
+
+	// Add the delegation to the repository
+	err = nRepo.RemoveDelegation(role)
+	if err != nil {
+		return fmt.Errorf("failed to remove delegation: %v", err)
+	}
+	cmd.Println("")
+	cmd.Printf(
+		"Removal of delegation of key \"%s\" to role %s, to repository \"%s\" staged for next publish.\n",
+		keyID, role, gun)
+	cmd.Println("")
+
+	return nil
+}
+
+// delegationAdd adds a public key from a certificate to a specific role in a GUN
+func (d *delegationCommander) delegationAdd(cmd *cobra.Command, args []string) error {
+	if len(args) < 4 {
+		return fmt.Errorf("must specify the Global Unique Name, the public key certificate path, the role of the delegation to add and a list of paths")
+	}
+
+	config := d.configGetter()
+
+	gun := args[0]
+	pubKeyPath := args[1]
+	role := args[2]
+	paths := args[3:]
+
+	// Check to see if the public key file exists
+	if _, err := os.Stat(pubKeyPath); os.IsNotExist(err) {
+		return fmt.Errorf("pub key file does not exist: %s", pubKeyPath)
+	}
+
+	// Read public key bytes from PEM file
+	pubKeyBytes, err := ioutil.ReadFile(pubKeyPath)
+	if err != nil {
+		return fmt.Errorf("unable to read public key from file: %s", pubKeyPath)
+	}
+
+	// Parse PEM bytes into type PublicKey
+	pubKey, err := trustmanager.ParsePEMPublicKey(pubKeyBytes)
+	if err != nil {
+		return fmt.Errorf("unable to parse valid public key certificate from PEM file %s: %v", pubKeyPath, err)
+	}
+
+	keyID := pubKey.ID()
+
+	// Check to see if it is an invalid ID
+	if len(keyID) != idSize {
+		return fmt.Errorf("invalid key ID provided: %s", keyID)
+	}
+
+	// no online operations are performed by add so the transport argument
+	// should be nil
+	nRepo, err := notaryclient.NewNotaryRepository(config.GetString("trust_dir"), gun, getRemoteTrustServer(config), nil, retriever)
+	if err != nil {
+		fatalf(err.Error())
+	}
+
+	// Add the delegation to the repository
+	err = nRepo.AddDelegation(role, 1, []data.PublicKey{pubKey}, paths)
```

Comment:
This is the number of thresholds, I can definitely make a const to make things clearer.  We could also try to make this configurable, though I don't think we've surfaced a way to add keys to non-root roles or configure role thresholds at the moment